src/Stratis.Bitcoin.Tests/BlockPulling2/BlockPullerBehaviorTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using Moq;
+using Stratis.Bitcoin.BlockPulling2;
+using Stratis.Bitcoin.Configuration.Logging;
+using Stratis.Bitcoin.Interfaces;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.BlockPulling2
+{
+    public class BlockPullerBehaviorTests
+    {
+        private readonly BlockPullerBehavior behavior;
+
+        public BlockPullerBehaviorTests()
+        {
+            var puller = new Mock<IBlockPuller>();
+
+            var ibdState = new Mock<IInitialBlockDownloadState>();
+            ibdState.Setup(x => x.IsInitialBlockDownload()).Returns(() => true);
+
+            var loggerFactory = new ExtendedLoggerFactory();
+            loggerFactory.AddConsoleWithFilters();
+
+            this.behavior = new BlockPullerBehavior(puller.Object, ibdState.Object, loggerFactory);
+        }
+
+        [Fact]
+        public void InitializesWithDefaultValues()
+        {
+            Assert.Equal(BlockPullerBehavior.SamplelessQualityScore, this.behavior.QualityScore);
+            Assert.Equal(0, this.behavior.SpeedBytesPerSecond);
+            Assert.Null(this.behavior.Tip);
+        }
+
+        [Fact]
+        public void QualityScoreCanGoToMinimum()
+        {
+            this.behavior.Penalize(10, 10);
+            this.behavior.RecalculateQualityScore(1000);
+
+            Assert.Equal(BlockPullerBehavior.MinQualityScore, this.behavior.QualityScore);
+        }
+
+        [Fact]
+        public void QualityScoreCanGoToMaximum()
+        {
+            this.behavior.AddSample(1000, 1);
+            this.behavior.RecalculateQualityScore(1);
+
+            Assert.Equal(BlockPullerBehavior.MaxQualityScore, this.behavior.QualityScore);
```

Comment:
Double?

Reply:
Same comment as before. 
Here double will be always exactly equal to MinQualityScore